extensions/amp-accordion/1.0/accordion.js

```diff
@@ -397,16 +415,47 @@ export function AccordionContent({
 }) {
   const ref = useRef(null);
   const hasMountedRef = useRef(false);
-  const {contentId, headerId, expanded, animate, setContentId} = useContext(
-    SectionContext
-  );
+  const {
+    contentId,
+    headerId,
+    expanded,
+    animate,
+    setContentId,
+    expandHandler,
+    experimentDisplayLocking,
+  } = useContext(SectionContext);
   const classes = useStyles();
+  const [contentVisibility, setContentVisibility] = useState(false);
+  const useDisplayLocking = contentVisibility && experimentDisplayLocking;
 
   useEffect(() => {
     hasMountedRef.current = true;
     return () => (hasMountedRef.current = false);
   }, []);
 
+  useEffect(() => {
+    if (!ref.current) {
+      return;
+    }
+    const element = ref.current;
+    const win = element.ownerDocument.defaultView;
+    const supportsContentVisibility = win.CSS.supports(
+      'content-visibility',
+      'hidden-matchable'
+    );
+    setContentVisibility(supportsContentVisibility);
+    if (!experimentDisplayLocking || !supportsContentVisibility) {
```

Comment:
We can move the `!experimentDisplayLocking` guard with the earliest `return` condition.

Reply:
I agree the component will do the same thing in the end.  One benefit of having it where it is though is that the `contentVisibility` state variable will be set to the correct value.  Ultimately, we need both the `contentVisibility` and `experimentDisplayLocking` to be true for the component to turn on the display locking behavior so regardless the final behavior will be the same, but I do like keeping the `contentVisibility` state variable in sync with the environment.  Thoughts?

Reply:
Calling `setContentVisibility` without reading the `contentVisibility` state will cause an unnecessary render. Not a big deal but something to be aware of. If we flip `experimentDisplayLocking` then this effect will run again and ensure the proper value is set at that point. I don't feel strongly though.

Reply:
Sounds good.